### PR TITLE
Separate format check

### LINKED ISSUE #4: Format error shouldn't prevent tests from running
What's the rationale behind stopping all tests if the format is wrong? My dev workflow is often incremental, thus I do a little thing, push, then come back later once tests have run. If the only thing missing the formatting while all functional tests are green, I usually just apply the formatter and immediately request a review.

IMHO, we should either make the format check a separate run on a single OS, or move it to the end of the tests. I prefer the former. Thoughts @efaulhaber @ranocha ?

### PR DESCRIPTION
**Pull request by sloede**
Resolves #4.

### CODE REVIEW

File: .github/workflows/FormatCheck.yml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        julia-version: [1]
+        os: [ubuntu-latest]
+    steps:
+      - uses: julia-actions/setup-julia@latest
+        with:
+          version: ${{ matrix.julia-version }}
+
```
Comment (efaulhaber) [inline .github/workflows/FormatCheck.yml:21]:
```suggestion
```

File: .github/workflows/FormatCheck.yml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+      - uses: actions/checkout@v4
+      - name: Install JuliaFormatter and format
+        # This will use the latest version by default but you can set the version like so:
+        #
+        # julia  -e 'using Pkg; Pkg.add(PackageSpec(name = "JuliaFormatter", version = "0.13.0"))'
+        #
+        # TODO: Change the call below to
+        #       format(".")
```
Comment (efaulhaber) [inline .github/workflows/FormatCheck.yml:29]:
```suggestion
```

File: .github/workflows/FormatCheck.yml [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+        run: |
+          julia -e '
+          out = Cmd(`git diff --name-only`) |> read |> String
+          if out == ""
+              exit(0)
+          else
+              @error "Some files have not been formatted !!!"
+              write(stdout, out)
+              exit(1)
+          end'
```
Comment (efaulhaber) [inline .github/workflows/FormatCheck.yml:43]: Why so complicated?
```suggestion
          julia -e 'using JuliaFormatter; !format(".") && error("Code not properly formatted")'
```

**Review by efaulhaber — CHANGES_REQUESTED** [review]

File: .github/workflows/FormatCheck.yml (hunk 3 above)
Comment (sloede) [reply .github/workflows/FormatCheck.yml:39]: IIRC, my (actually @ranocha's) version prints _only_ the files that have not been formatted, which is nice imho

File: .github/workflows/FormatCheck.yml [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    strategy:
+      matrix:
+        julia-version: [1]
+        os: [ubuntu-latest]
+    steps:
+      - uses: julia-actions/setup-julia@latest
+        with:
+          version: ${{ matrix.julia-version }}
+      - uses: actions/checkout@v4
+      - name: Install JuliaFormatter and format
```
Comment (ranocha) [inline .github/workflows/FormatCheck.yml:22]: Caching?

Comment (sloede) [reply .github/workflows/FormatCheck.yml]: Thanks, added in 44a16f9

**Review by ranocha — APPROVED** [review]

File: .github/workflows/FormatCheck.yml (hunk 3 above)
Comment (sloede) [reply .github/workflows/FormatCheck.yml:43]: I'll go ahead and resolve this then.

**Comment by sloede** [comment]
@efaulhaber it would be great to get your approval as well such that the auto-merge cna go ahead

**Review by efaulhaber — APPROVED** [review]

**Pull request merged**